modules/contact/test/tests/fieldsplit/tests

```diff
@@ -1,11 +1,19 @@
 [Tests]
-  issues = '#7485'
-  design = 'FieldSplitPreconditioner.md'
+  issues = '#7485 #14791'
+  design = 'FieldSplitPreconditioner.md NormalMortarLMMechanicalContact.md TangentialMortarLMMechanicalContact.md'
   [./2block-3D]
     type = 'Exodiff'
     input = '2blocks3d.i'
     exodiff = '2blocks3d_out.e'
     max_parallel = 4
     requirement = 'The system shall allow for split preconditioning based on contact surfaces.'
   [../]
+  [./mortar-FS]
+    type = 'Exodiff'
+    input = 'frictional_mortar_FS.i'
+    exodiff = 'frictional_mortar_FS_out.e'
+    abs_zero = 1e-8
+    max_parallel = 1
```

Comment:
Is there a particular reason that the code does not run in parallel?

Reply:
It did not pass some of the parallel moosebuild tests so I made this to run only in serial. I found the test error was due to some tolerance settings, and I will update the input and tests file correspondingly.

Reply:
Yea I would definitely hope that this field split capability allows us to run in parallel otherwise I see little advantage over using LU (which definitely works in parallel)

Reply:
> It did not pass some of the parallel moosebuild tests so I made this to run only in serial. I found the test error was due to some tolerance settings, and I will update the input and tests file correspondingly.

👍

Reply:
> Yea I would definitely hope that this field split capability allows us to run in parallel otherwise I see little advantage over using LU (which definitely works in parallel)

Yea I agree! Just FYI, I ran some cases to compare the performances of different preconditioners for mortar contact problems (in serial). It looks like FS performs better than LU for relatively small problem sizes, but cannot compete super LU (not very surprising...). I would suppose FS performs and scales better for large problems in parallel, and that would be my next task.

Reply:
Better in terms of compute time or iteration?

Reply:
> Better in terms of compute time or iteration?

FS is better in terms of total Flop.

Reply:
Yea, so by what metric is LU better?

Reply:
> Yea, so by what metric is LU better?

I choose Flop as the only metric for now (should be consistent with compute time). Also, I only have serial comparisons. Under this setting, FS performs better than LU (but not superLU) for problem sizes around 1e3 elements. However, I have yet to examine the parallel performance. The hope is that FS -- as an iterative preconditioner -- should perform much better than LU (hopefully superLU, too) when run in parallel. I will keep you posted on this :)

Reply:
> Under this setting, FS performs better than LU (but not superLU) for problem sizes around 1e3 elements

So when running in serial, for 1000 elements, performance is like this: SuperLU is better than FSP which is better than PETSc default LU?

Reply:
Could you try running with like 100,000 elements (in serial)? @fdkong and I hope that at that point FSP should be more performant than SuperLU. If you want, you can come visit the MOOSE cube. We are both here this week

Reply:
> So when running in serial, for 1000 elements, performance is like this: SuperLU is better than FSP which is better than PETSc default LU?

Yes, that is correct for mortar contact.

Reply:
> Could you try running with like 100,000 elements (in serial)? @fdkong and I hope that at that point FSP should be more performant than SuperLU. If you want, you can come visit the MOOSE cube. We are both here this week.
> 
> I don't know how many time steps you're running...100,000 elements in serial may take too long...

I would expect 1e5 elements being too large for a serial run. This is also why I would like to try comparing parallel performances. I am currently occupied with something else, but yea, I will visit you guys later this week in order to move this forward!